locale.c

```diff
@@ -5881,48 +5888,68 @@ the program, giving results based on that locale.
 
 =cut
  */
-    PERL_ARGS_ASSERT_MY_STRFTIME;
 
-    /* An empty format yields an empty result */
-    const int fmtlen = strlen(fmt);
-    if (fmtlen == 0) {
-        char *ret;
-        Newxz (ret, 1, char);
-        return ret;
-    }
+STATIC struct tm *
+S_ints_to_tm(pTHX_ int sec, int min, int hour, int mday, int mon, int year,
+                   int wday, int yday, int isdst)
+{
+    /* Create a struct tm structure from the input time-related integer
+     * variables */
 
     /* Set mytm to now */
-    struct tm mytm;
-    init_tm(&mytm);	/* XXX workaround - see Perl_init_tm() */
+    struct tm * mytm;
+    Newxz(mytm, 1, struct tm);
+    init_tm(mytm);	/* XXX workaround - see Perl_init_tm() */
 
     /* Override with the passed-in values */
-    mytm.tm_sec = sec;
-    mytm.tm_min = min;
-    mytm.tm_hour = hour;
-    mytm.tm_mday = mday;
-    mytm.tm_mon = mon;
-    mytm.tm_year = year;
-    mytm.tm_wday = wday;
-    mytm.tm_yday = yday;
-    mytm.tm_isdst = isdst;
-    mini_mktime(&mytm);
+    mytm->tm_sec = sec;
+    mytm->tm_min = min;
+    mytm->tm_hour = hour;
+    mytm->tm_mday = mday;
+    mytm->tm_mon = mon;
+    mytm->tm_year = year;
+    mytm->tm_wday = wday;
+    mytm->tm_yday = yday;
+    mytm->tm_isdst = isdst;
+    mini_mktime(mytm);
 
     /* use libc to get the values for tm_gmtoff and tm_zone on platforms that
      * have them [perl #18238] */
-#  if  defined(HAS_MKTIME)                                      \
-   && (defined(HAS_TM_TM_GMTOFF) || defined(HAS_TM_TM_ZONE))
+#if  defined(HAS_MKTIME)                                      \
+ && (defined(HAS_TM_TM_GMTOFF) || defined(HAS_TM_TM_ZONE))
     struct tm mytm2;
-    mytm2 = mytm;
+    StructCopy(mytm, &mytm2, struct tm);
+    mytm2 = *mytm;
```

Comment:
These two lines do the same thing, though the second is safer.

Reply:
Yes; I forgot to delete the first line.  Will fix

Reply:
fixed